app/src/main/java/ch/hikemate/app/model/authentication/FirebaseAuthRepository.kt

```diff
@@ -80,6 +86,13 @@ class FirebaseAuthRepository {
             onErrorAction(task.exception ?: Exception("Unknown error"))
           }
         }
+      } catch (e: NoCredentialException) {
+        // In case no credentials are found, show a toast message and invoke the error callback
+        Log.d("SignInButton", "No credentials found: ${e.message}")
+        CoroutineScope(Dispatchers.Main).launch {
+          Toast.makeText(context, "No Google account found on the device", Toast.LENGTH_LONG).show()
+        }
+        onErrorAction(e)
```

Comment:
I like the way you handle the error. But I think display a Toast should not be in the repository as this is related to UI.
I don't think you can do it in the Screen as of how the architecture is but you could do it in the view model.

Reply:
Yep, I fully agree, I'm editing it

Reply:
Also, is this the official way to handle it ? By that I mean, is it mandatory to have a Google account link to the device to be able to sign in with Google ?